### DISCUSSION TITLE
Raw SQL Queries in Insights Module

### DISCUSSION THREAD

**Discussion by etmartinkazoo** [directus/directus#14558] [Feature Requests]
I love the Insights module, it is already a phenomenal tool. It would be even better if one could write raw SQL queries as it would give even more power to something like "Metrics" where I need to get specific about time frames.

Does anyone have any ideas about doing this?

**Comment by rijkvanzanten** [comment]
I don't think we'll expose actual raw SQL from the app, as that would break the cross database compatibility (eg it would lock you to a certain db vendor), an is a bit of a security risk. I think the second best way to achieve something like that would be with a custom panel w/ a custom endpoint to return the data in exactly the way you need it 🤔

**Reply by etmartinkazoo** [reply]
That makes good sense. Perhaps an extension would work so that developers who want/need the flexibility have the option without baking it into core.

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
